js/src/offcanvas.js

```diff
@@ -0,0 +1,239 @@
+  constructor(element) {
+    super(element)
+
+    this._isShown = element.classList.contains(CLASS_NAME_SHOW)
+    this._bodyOptions = element.getAttribute(DATA_BODY_ACTIONS) || ''
+    this._handleClosing()
+  }
+
+  // Public
+
+  toggle(relatedTarget) {
+    return this._isShown ? this.hide() : this.show(relatedTarget)
+  }
+
+  show(relatedTarget) {
+    if (this._isShown) {
+      return
+    }
+
+    const showEvent = EventHandler.trigger(this._element, EVENT_SHOW, { relatedTarget })
+
+    if (showEvent.defaultPrevented) {
+      return
+    }
+
+    this._isShown = true
+    this._element.style.visibility = 'visible'
+
+    if (this._bodyOptionsHas('backdrop') || !this._bodyOptions.length) {
+      document.body.classList.add(CLASS_NAME_BACKDROP_BODY)
+    }
+
+    if (!this._bodyOptionsHas('scroll')) {
+      scrollBarHide()
+    }
+
+    this._element.classList.add(CLASS_NAME_TOGGLING)
+    this._element.removeAttribute('aria-hidden')
+    this._element.setAttribute('aria-modal', true)
+    this._element.setAttribute('role', 'dialog')
+    this._element.classList.add(CLASS_NAME_SHOW)
+
+    const completeCallBack = () => {
+      this._element.classList.remove(CLASS_NAME_TOGGLING)
+      EventHandler.trigger(this._element, EVENT_SHOWN, { relatedTarget })
+      this._enforceFocusOnElement(this._element)
+    }
+
+    setTimeout(completeCallBack, getTransitionDurationFromElement(this._element))
+  }
+
+  hide() {
+    if (!this._isShown) {
+      return
+    }
+
+    const hideEvent = EventHandler.trigger(this._element, EVENT_HIDE)
+
+    if (hideEvent.defaultPrevented) {
+      return
+    }
+
+    this._element.classList.add(CLASS_NAME_TOGGLING)
+    EventHandler.off(document, EVENT_FOCUSIN)
+    this._element.blur()
+    this._isShown = false
+    this._element.classList.remove(CLASS_NAME_SHOW)
+
+    const completeCallback = () => {
+      this._element.setAttribute('aria-hidden', true)
+      this._element.removeAttribute('aria-modal')
+      this._element.removeAttribute('role')
+      this._element.style.visibility = 'hidden'
+
+      if (this._bodyOptionsHas('backdrop') || !this._bodyOptions.length) {
+        document.body.classList.remove(CLASS_NAME_BACKDROP_BODY)
+      }
+
+      if (!this._bodyOptionsHas('scroll')) {
+        scrollBarReset()
+      }
+
+      EventHandler.trigger(this._element, EVENT_HIDDEN)
+      this._element.classList.remove(CLASS_NAME_TOGGLING)
+    }
+
+    setTimeout(completeCallback, getTransitionDurationFromElement(this._element))
+  }
+
+  _enforceFocusOnElement(element) {
+    EventHandler.off(document, EVENT_FOCUSIN) // guard against infinite focus loop
+    EventHandler.on(document, EVENT_FOCUSIN, event => {
+      if (document !== event.target &&
+        element !== event.target &&
+        !element.contains(event.target)) {
+        element.focus()
+      }
+    })
+    element.focus()
+  }
+
+  _bodyOptionsHas(option) {
+    return this._bodyOptions.split('|').includes(option)
```

Comment:
Few plugins are already using comma separated values, so if you also use a comma here instead of a pipe, that would keep consistency IMO.

Reply:
I agree and I 've changed it. IMO I would prefer 'pipe'. Seems more programmatic in my eyes